scripts/tests/run_tv_casting_test.py

```diff
@@ -0,0 +1,305 @@
+        self.mode = mode
+
+    # Enter the context manager to open and return the log file.
+    def __enter__(self):
+        try:
+            self.file = open(self.log_file_path, self.mode)
+        except FileNotFoundError:
+            # Handle file not found error
+            raise FileNotFoundError(f"Log file '{self.log_file_path}' not found.")
+        except IOError:
+            # Handle IO error
+            raise IOError(f"Error opening log file '{self.log_file_path}'.")
+        return self.file
+
+    # Exit the context manager, closing and removing the log file created.
+    #   exception_type: The type of exception that occurred, if any.
+    #   exception_value: The value of the exception, if any.
+    #   traceback: The traceback of the exception.
+    def __exit__(self, exception_type, exception_value, traceback):
+        self.file.close()
+
+        if os.path.exists(self.log_file_path):
+            os.remove(self.log_file_path)
+
+
+class ProcessManager:
+    """
+    A context manager for managing subprocesses.
+
+    This class provides a context manager for safely starting and stopping a subprocess.
+    """
+
+    # Initialize ProcessManager.
+    #   command (list): The command to execute as a subprocess.
+    #   stdout (file): File-like object to which the subprocess's standard output will be redirected.
+    #   stderr (file): File-like object to which the subprocess's standard error will be redirected.
+    def __init__(self, command: List[str], stdout, stderr):
+        self.command = command
+        self.stdout = stdout
+        self.stderr = stderr
+
+    # Enter the context manager to start the subprocess and return it.
+    def __enter__(self):
+        self.process = subprocess.Popen(self.command, stdout=self.stdout, stderr=self.stderr, text=True)
+        return self.process
+
+    # Exit the context manager, terminating the subprocess.
+    #   exception_type: The type of exception that occurred, if any.
+    #   exception_value: The value of the exception, if any.
+    #   traceback: The traceback of the exception.
+    def __exit__(self, exception_type, exception_value, traceback):
+        self.process.terminate()
+        self.process.wait()
+
+
+# Dump the contents of a log file to the console.
+#   log_file_path: The path to the log file.
+def dump_logs_to_console(log_file_path: str):
+    print('\nDumping logs from: ', log_file_path)
+
+    with LogFileManager(log_file_path, 'r') as file:
+        for line in file:
+            print(line.rstrip())
+
+
+# Log 'Discovery failed!' as an error, dump the contents of the log files
+# to the console, exit on error.
+#   log_file_paths: A list of paths to the log files, i.e. the path to the
+#                   Linux tv-casting-app logs and the tv-app logs.
+def handle_discovery_failure(log_file_paths: List[str]):
+    logging.error('Discovery failed!')
+
+    for log_file_path in log_file_paths:
+        dump_logs_to_console(log_file_path)
```

Comment:
nit: technically this will leave some files not removed if a dump of several files fails in the middle. For example if I dump files A,B,C and opening B fails for some reason, then we have:
  - dump A, remove A
  - start dump B ... throw
  
then once this handle exits, we both do not `sys.exit` and file C is not even attempted to be removed.

Maybe we should wrap these dumps into a try/except with a `logging.exception(f"Failed to dump {log_file_path}")` to prevent this.

Reply:
Great suggestion! The change has been added to the latest commit.